woocommerce/woocommerce
# Zoom improvements for mobile/touch

### All Submissions:


### Changes proposed in this Pull Request:

This PR fixes #19959 by reverting PR #19954 and approaching #19688 differently.

Detecting zoom support is not reliable and affects both mobile and touch screen laptops.

Listening for touch support works but requires interaction first.

We should ensure the gallery works across all devices in that case.

Rather than disable zoom on mobile this:

- Correctly implements click to zoom which zooms to a specific area on click when using mobile.
- Updates the zoom script which has some fixes/accessibility improvements
- Ensures swipe works without trigger photoswipe

### How to test the changes in this Pull Request:

Have a product with a gallery containing multiple items.

On desktop:

1. Click the icon to open lightbox
2. Hover to zoom
3. Ensure images switch with the gallery controls
4. Ensure zoom is correct after the image changes

On mobile or iOS simulator:

1. Click the icon to open lightbox
2. Click the image and click again to zoom in/out
3. Ensure images switch with the gallery controls
4. Ensure zoom is correct after the image changes
5. Ensure swipe to change image still functions
6. Ensure swipe to scoll the page still functions even if you swipe from the featured image.

cc @jameskoster
